src/Compilers/CSharp/Portable/Binder/Binder_Patterns.cs

```diff
@@ -829,7 +795,7 @@ private BoundPattern BindVarDesignation(VarPatternSyntax node, VariableDesignati
                             {
                                 var location = new SourceLocation(node.SyntaxTree, 
                                     new Text.TextSpan(tupleDesignation.OpenParenToken.SpanStart, tupleDesignation.CloseParenToken.Span.End - tupleDesignation.OpenParenToken.SpanStart));
-                                diagnostics.Add(ErrorCode.ERR_WrongNumberOfSubpatterns, location, inputType.TupleElementTypes, elementTypes.Length, tupleDesignation.Variables.Count);
+                                diagnostics.Add(ErrorCode.ERR_WrongNumberOfSubpatterns, location, inputType.ToDisplayString(), elementTypes.Length, tupleDesignation.Variables.Count);
```

Comment:
>.ToDisplayString() [](start = 107, length = 18)

Symbols shouldn't be eagerly replaced with strings when diagnostics is constructed. If default formatting doesn't work for the scenario, ```FormattedSymbol``` helper can be used. There are many examples of that in the codebase.